packages/playground-server/src/middleware/create-account.ts

```diff
@@ -0,0 +1,115 @@
+import { verifyMessage } from "ethers/utils";
+import { Context } from "koa";
+import "koa-body"; // See: https://github.com/dlau/koa-body/issues/109
+
+import { createMultisigFor } from "../node";
+import {
+  ApiResponse,
+  CreateAccountRequest,
+  CreateAccountResponseData,
+  ErrorCode
+} from "../types";
+
+function buildSignaturePayload(data: CreateAccountRequest) {
+  return [
+    "PLAYGROUND ACCOUNT REGISTRATION",
+    `Username: ${data.username}`,
+    `E-mail: ${data.email}`,
+    `Ethereum address: ${data.address}`
+  ].join("\n");
+}
+
+function validateRequest(params: CreateAccountRequest): ApiResponse {
+  if (!params.username) {
+    return {
+      ok: false,
+      error: {
+        status: 400,
+        errorCode: ErrorCode.UsernameRequired
+      }
+    };
+  }
+
+  if (!params.email) {
+    return {
+      ok: false,
+      error: {
+        status: 400,
+        errorCode: ErrorCode.EmailRequired
+      }
+    };
+  }
+
+  if (!params.address) {
+    return {
+      ok: false,
+      error: {
+        status: 400,
+        errorCode: ErrorCode.AddressRequired
+      }
+    };
+  }
+
+  if (!params.signature) {
+    return {
+      ok: false,
+      error: {
+        status: 400,
+        errorCode: ErrorCode.SignatureRequired
+      }
+    };
+  }
+
+  const providedSignature = params.signature;
+  const expectedMessage = buildSignaturePayload(params);
+  const expectedAddress = verifyMessage(expectedMessage, providedSignature);
+
+  if (
+    // We compare the addresses case-insensitively.
+    // verifyMessage() returns mixed upper and lower case characters,
+    // while eth.accounts[0] always returns lower-case characters.
+    params.address.localeCompare(expectedAddress, "en", {
```

Comment:
I believe you can use `ethers` here: https://docs.ethers.io/ethers.js/html/api-utils.html#addresses

```js
getAddress(params.address) === expectedAddress
```

Reply:
@snario Great, this is way better. I'll implement it right away :)